Packs/MicrosoftGraphApplications/Integrations/MicrosoftGraphApplications/MicrosoftGraphApplications.py

```diff
@@ -152,6 +229,134 @@ def list_service_principals_command(ms_client: Client, args: dict) -> CommandRes
     )
 
 
+def validate_service_principal_input(args: dict) -> tuple:
+    """
+    Ensure at least one argument is given.
+
+    Args:
+        args: The arguments were passed with the command.
+
+    Returns:
+        If the two arguments are missing, raise an exception, otherwise return one of them.
+    """
+    object_id = args.get('id')
+    app_client_id = args.get('app_id')
+    if not (object_id or app_client_id):
+        raise DemistoException("User must provide one of (object) id or application id.")
+
+    # if both are provided, pass the object_id
+    if object_id:
+        return f"/{object_id}", object_id
+    return f"(appId='{app_client_id}')", app_client_id
+
+
+def get_service_principal_command(ms_client: Client, args: dict) -> CommandResults:
+    """
+
+    Args:
+        ms_client: The Client
+        args: demisto.args()
+
+    Returns:
+        Results to post in demisto
+    """
+    id_for_request, service_id = validate_service_principal_input(args=args)
+    results = ms_client.get_single_service_principal(id_for_request)
+    return CommandResults(
+        'MSGraphApplication',
+        'id',
+        outputs=results,
+        readable_output=tableToMarkdown(
+            'Available service (application):',
+            results,
+            headers=['id', 'appId', 'appDisplayName', 'accountEnabled', 'deletedDateTime'],
+            removeNull=True
+        )
+    )
+
+
+def update_service_principal_command(ms_client: Client, args: dict) -> CommandResults:
+    """
+    Update the properties of servicePrincipal object.
+    Args:
+        ms_client: The Client
+        args: demisto.args()
+
+    Returns:
+        an informative message.
+    """
+    id_for_request, service_id = validate_service_principal_input(args=args)
+    data = {}
+
+    if account_enabled := args.get("account_enabled"):
+        data["accountEnabled"] = argToBoolean(account_enabled)
+
+    # Dict
+    if add_ins := args.get("add_ins"):
+        data["addIns"] = add_ins
+
```

Comment:
repetitive code can be cleaner with a simple loop, and a list of origin keys (e.g. `alternative_names`) and destination keys (e.g. `alternativeName`), maybe two loops - one for those on which we want to run argToBoolean and one for the rest.

Reply:
Done